04/app.js

```diff
@@ -1,89 +1,117 @@
+import React from "react";
+import { createRoot } from "react-dom/client";
 
-const root = createRoot(document.querySelector('#root'));
+const root = createRoot(document.querySelector("#root"));
 
 class App extends React.Component {
-    state = { 
-        firstName: '',
-        lastName: '',
-        searchQuery: '',
-        users: ['Jan Kowalski', 'Michał Nowak'],
-    }
+  state = {
+    firstName: "",
+    lastName: "",
+    searchQuery: "",
+    users: ["Jan Kowalski", "Michał Nowak"],
+  };
 
-    renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
-            return (
-                <li onClick={ this.clickHandler }>
-                    { name }
-                </li>
-            );
-        });
+  renderUsersList() {
+    const { users, searchQuery } = this.state;
+    if (searchQuery) {
+      return users.map((user) => {
+        if (user.toLowerCase().includes(searchQuery.toLowerCase())) {
+          return <li onClick={this.clickHandler}>{user}</li>;
+        }
+      });
     }
+    return users.map((name) => {
+      return <li onClick={this.clickHandler}>{name}</li>;
+    });
+  }
 
-    clickHandler = e => {
-        const {innerText: userName} = e.target;
-        this.removeUser(userName);
-    }
+  clickHandler = (e) => {
+    const { innerText: userName } = e.target;
+    this.removeUser(userName);
+  };
 
-    inputChange = e => {
-        const {name, value} = e.target;
-        this.setState({
-            [name]: value,
-        });
-    }
+  inputChange = (e) => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value,
+    });
+  };
 
-    render() {
-        const { firstName, lastName } = this.state;
-        return (
-            <section onSubmit={ this.submitHandler }>
-                <form>
-                    <input name="firstName"
-                        value={ firstName }
-                        onChange={ this.inputChange }
-                    />
-                    <input name="lastName"
-                        value={ lastName }
-                        onChange={ this.inputChange }
-                    />
-                    <input type="submit"/>
-                </form>
-                <ul>{ this.renderUsersList() }</ul>
-            </section>
-        );
+  render() {
+    const { firstName, lastName } = this.state;
+    return (
+      <section onSubmit={this.submitHandler}>
+        <form>
+          <input
+            name="firstName"
+            value={firstName}
+            onChange={this.inputChange}
+          />
+          <input
+            name="lastName"
+            value={lastName}
+            onChange={this.inputChange}
+          />
+          <input type="submit" />
+        </form>
+        <ul>{this.renderUsersList()}</ul>
+        <input
+          name="searchQuery"
+          onChange={this.inputChange}
+        ></input>
+      </section>
+    );
+  }
+
+  searchHandler = (e) => {
```

Comment:
Jeśli nie korzystasz już  z tej funkcji to trzeba ja usunąć.
Ten kod nie wygląda "zbyt ładnie" - mimo, że z niego nie korzystać to pewnie rekruter zwrócił bo na niego uwagę ;)